cms/djangoapps/contentstore/signals/handlers.py

```diff
@@ -43,6 +51,48 @@ def wrapper(*args, **kwargs):
     return task_decorator
 
 
+def create_catalog_data_for_signal(course_key: CourseKey) -> Optional[CourseCatalogData]:
+    """
+    Creates data for catalog-info-changed signal when course is published.
+
+    Arguments:
+        course_key: Key of the course to announce catalog info changes for
+
+    Returns:
+        Data for signal, or None if not appropriate to send on this signal.
+    """
+    # Only operate on real courses, not libraries.
+    if not course_key.is_course:
+        return None
+
+    store = modulestore()
+    with store.branch_setting(ModuleStoreEnum.Branch.published_only, course_key):
+        course = store.get_course(course_key)
+
+        return CourseCatalogData(
+            course_key=course_key.for_branch(None),
+            name=course.display_name,
+            schedule_data=CourseScheduleData(
+                start=course.start,
+                pacing='self' if course.self_paced else 'instructor',
+                end=course.end,
+                enrollment_start=course.enrollment_start,
+                enrollment_end=course.enrollment_end,
+            ),
+            effort=CourseDetails.fetch_about_attribute(course_key, 'effort'),
+            hidden=course.catalog_visibility in ['about', 'none'] or course.id.deprecated,
```

Comment:
Is this `course.id` the same as the `course_key` we already have? I couldn't figure out what `get_course` even returns...

Reply:
Yes, `course.id` should return a `CourseKey` and they should match.